src/client/pydaos/pydaos_core.py

```diff
@@ -154,6 +159,30 @@ def rootkv(self, cid=ObjClassID.OC_SX):
         oid = ObjID(hi, lo)
         return KVObj(self.coh, oid, self)
 
+    def get_kv_by_name(self, name, root=None, create=False):
+        """Return KV by name.
+
+        Allow selection of root (or parent) container, and
+        optionally create kv if not found"""
+
+        if not root:
+            root = self.rootkv()
+        if name in root:
+            object_data = pickle.loads(root[name])
+            return self.kv(object_data['oid'])
+
+        if not create:
+            raise KvNotFound
+
+        new_kv = self.newkv()
+        # Create a new entry in the root kv, where the entry
+        # itself is a dick, and the 'oid' entry is the object
```

Comment:
I think you've got a typo here where you meant 'dict'...

Reply:
I think I probably did.

Reply:
:)